packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -64,29 +68,70 @@ export function getBeaconBlockApi({
 
     // check what validations have been requested before broadcasting and publishing the block
     // TODO: add validation time to metrics
-    const broadcastValidation = opts.broadcastValidation ?? routes.beacon.BroadcastValidation.none;
+    const broadcastValidation = opts.broadcastValidation ?? routes.beacon.BroadcastValidation.gossip;
     // if block is locally produced, full or blinded, it already is 'consensus' validated as it went through
     // state transition to produce the stateRoot
     const slot = signedBlock.message.slot;
+    const fork = config.getForkName(slot);
     const blockRoot = toHex(chain.config.getForkTypes(slot).BeaconBlock.hashTreeRoot(signedBlock.message));
     const blockLocallyProduced =
       chain.producedBlockRoot.has(blockRoot) || chain.producedBlindedBlockRoot.has(blockRoot);
     const valLogMeta = {broadcastValidation, blockRoot, blockLocallyProduced, slot};
 
     switch (broadcastValidation) {
-      case routes.beacon.BroadcastValidation.none: {
-        if (blockLocallyProduced) {
-          chain.logger.debug("No broadcast validation requested for the block", valLogMeta);
-        } else {
-          chain.logger.warn("No broadcast validation requested for the block", valLogMeta);
+      case routes.beacon.BroadcastValidation.gossip: {
+        if (!blockLocallyProduced) {
+          try {
+            await validateGossipBlock(config, chain, signedBlock, fork);
+          } catch (error) {
+            chain.logger.error("Gossip validations failed while publishing the block", valLogMeta, error as Error);
+            throw error;
+          }
         }
+        chain.logger.debug("Gossip checks validated while publishing the block", valLogMeta);
+        // TODO: figure out if consensus validation is required for gossip or not
```

Comment:
I think this todo can be removed. What we seems to align with the descriptions in the spec.